### ISSUE TITLE
Add support for ECDH-1PU+ XC20PKW

### ISSUE THREAD

**Issue by awoie** [decentralized-identity/didcomm-messaging#187]
Currently, the spec only supports AES-based key wrapping algorithms. IMO, we should have support for `XC20PKW` especially for people who use `XC20P` for content encryption to minimize crypto-libs.

`XC20PKW` is defined here: https://tools.ietf.org/html/draft-amringer-jose-chacha-02

**Comment by awoie** [comment]
We have our `ECDH-1PU+XC20PKW` implementation ([see impl in DIF did-jwt](https://github.com/awoie/did-jwt/blob/awoie/ecdh-1pu/src/xc20pEncryption.ts#L93-L95)) based on [Amringer's latest XC20PKW draft](https://tools.ietf.org/html/draft-amringer-jose-chacha-02) that only talks about Key Wrapping (in contrast to a previous version of the draft).

Please note, in the latest Amringer draft, there are `iv` and `tag` in the recipient header which is not the case in the most recent proposed Aries `ECDH-1PU+XC20PKW` test vectors which can be found [here](https://github.com/hyperledger/aries-rfcs/blob/master/features/0334-jwe-envelope/authcrypt-examples.md#214-x25519-keys). IMO, that is an independent problem that Aries 0334 needs to address. Note, since Amringer does not register `ECDH-1PU+XC20PKW`, Aries is not necessarily wrong but imo, it would make sense to use `XC20PKW` as per Amringer to avoid confusion.

A few things that I'd like to discuss:
1. Do we need to register `ECDH-1PU+XC20PKW ` in IANA and if this is the case, would this need to happen through Madden's ECDH-1PU draft?
2. If adopted by Madden, should `ECDH-1PU-XC20PKW` be based on Amringer's latest spec which is different from the test vector's in Aries? Note, there is no JWE spec that matches the XC20PKW in Aries yet. I'd say, we should try to keep inventing new specs minimal and yes, in that case Aries should likely use Amringer's latest spec. 
3. Someone probably needs to reach out to Amringer and ask about the status of the draft since it has not been adopted by CFRG yet and the draft has expired in July 2020. 

Any thoughts? @andrewwhitehead @oed @kdenhartog @TelegramSam @Baha-sk

**Comment by kdenhartog** [comment]
>Do we need to register ECDH-1PU+XC20PKW in IANA and if this is the case, would this need to happen through Madden's ECDH-1PU draft?

Yes, I suspect it will be faster to get XC20P through first before 1PU, so registering it in the 1PU draft makes more sense.

>If adopted by Madden, should ECDH-1PU-XC20PKW be based on Amringer's latest spec which is different from the test vector's in Aries?

It likely will have to be. I think the standards we can get moving will play the largest impact on this. I'm still waiting to hear back from CFRG chairs/Neil/Scott (xchacha20 rfc author) to see if this is something we can move forward.

**Comment by Baha-sk** [comment]
@kdenhartog @awoie, it seems to me adding new recipient headers for iv and tag is cumbersome and unnecessary.

I personally like how Tink returns cipher text from their primitives: they concatenate `nonce+ciphertext+tag` and return the result as a `ciphertext` to the user who doesn't need to know/parse `ciphertext`, the library takes care of it for them.

In Golang, the XChacha20Poly1305 primitive's Seal() call returns a ciphertext that is comprised of `ciphertext+tag` (Tink goes one step further by also prepending `nonce` to simplify the user's interaction with the library).

in AFGO, we've followed suit what Tink does. In our `ECDH-1PU-XC20PKW` [implementation](https://github.com/hyperledger/aries-framework-go/blob/main/pkg/crypto/tinkcrypto/wrap_support.go#L212), we made the `encrypted_key` header to contain the resulting `nonce+ciphertext+tag` value of the wrapped key. The recipient can extract them directly since both `nonce` and `tag` sizes are predetermined for XChacaha20Poly1305. 

I think adding `iv` and `tag` in the recipient headers provides additional info to a malicious user. There's no need to help them identify more info to break the encryption.
